### PR TITLE
rats-tls: divide into sub-packages

### PR DESCRIPTION
**Pull request by zhiminghufighting**
Fixes: #1512
Signed-off-by: zhiminghufighting <zhiming.hu@intel.com>

### CODE REVIEW

File: rats-tls/README.md [hunk 1]
```diff
@@ -51,11 +51,40 @@ make -C build install
...
+```shell
+make package MODE=sgx
+```
+
+If you want to build tdx mode related package, please type the following command.
+
+```shell
+make package MODE=tdx
+```
+
```
Comment (haosanzi) [inline rats-tls/README.md:87]: Thank you. 
I'm not sure if it's necessary to make the `make package` command support different modes.  
I think developers can call the `make package` command to generate packages in four modes: host, occlum, tdx, and sgx, and then maintain these packages to the IC repository. 
Users can install different packages according to their needs.

File: rats-tls/dist/rpm/rats_tls.spec.in [hunk 2]
```diff
@@ -51,12 +52,12 @@ if [ ! -d "$SGX_SDK" ]; then
         exit 1
 fi
 
-pushd %{name}
+pushd %{source_dir}
```
Comment (haosanzi) [inline rats-tls/dist/rpm/rats_tls.spec.in:55]: May I ask why change `name` to `source_dir`?

Comment (zhiminghufighting) [reply rats-tls/dist/rpm/rats_tls.spec.in:55]: Here build script needs to record the build path and popd it once cmd completed and the build path(rats-tls) is a constant string. But %{name} will be alternative with real package name according to different mode. That's why need use other available to record the constant path.

File: rats-tls/README.md (hunk 1 above)
Comment (zhiminghufighting) [reply rats-tls/README.md:87]: Do you mean the different package for each .so or .a?  The finally style will be that format but i don't think the style will be in such stage. Currently stage, 
In future mode, rats-tls after redesigned arch will be divided into three levels: core level/transport level/lib rats level, then we will make the install package according to different levels will make sense.

**Pull request closed without merging**